autocti/extract/two_d/parallel/abstract.py

```diff
@@ -42,74 +55,79 @@ def injection_norm_lists_from(
             FPR rows). To remove the 10 leading pixels which have lost electrons due to CTI, an input such
             as `pixels=(10, 30)` would be used.
         """
-        injection_norm_lists = []
+        median_list = []
 
         arr_list = [
             np.ma.array(data=array.native[region.slice], mask=array.mask[region.slice])
             for region in self.region_list_from(pixels=pixels)
         ]
 
-        for array_2d in arr_list:
-
-            injection_of_array_list = []
-
-            for column_index in range(array_2d.shape[1]):
+        for column_index in range(arr_list[0].shape[1]):
 
-                injection_norm = float(np.ma.median(array_2d[:, column_index]))
+            for i, array_2d in enumerate(arr_list):
 
-                injection_of_array_list.append(injection_norm)
+                if i == 0:
+                    arr = array_2d[:, column_index]
+                else:
+                    arr = np.concatenate((arr[:], array_2d[:, column_index]))

```

Comment:
This is super inefficient because you're allocating a new numpy array for each item in the loop. I think np.stack can be used to combine a list of numpy arrays

Reply:
Turns out it also had a bug, thanks!